### PR TITLE
Make status_code keyword-only

### PR DESCRIPTION
**Pull request by j178**
Make `status_code` a keyword-only argument, and defaulting it to 200. 

First part of #1227 .

### CODE REVIEW

File: tests/models/test_cookies.py [hunk 1]
```diff
@@ -69,7 +69,7 @@ def test_multiple_set_cookie():
         ),
     ]
     request = httpx.Request("GET", "https://www.example.org")
-    response = httpx.Response(200, request=request, headers=headers)
+    response = httpx.Response(status_code=200, request=request, headers=headers)
```
Comment (florimondmanca) [inline tests/models/test_cookies.py:72]: Do we want to keep these `status_code=200` in the sale of "explicit over implicit", or should we rely on the new "200 by default"?

**Review by florimondmanca — COMMENTED** [review]
Almost all good, just a bit question :)

**Comment by tomchristie** [comment]
Having reviewed the change footprint on this I'm thinking we *might* want to hold fire on this part? Looking at the test cases that change it's not so obviously a UX improvement. Once we've made `request=` optional, we could reassess?

File: tests/models/test_cookies.py (hunk 1 above)
Comment (j178) [reply tests/models/test_cookies.py:72]: I think explicitly specifying `status_code=200` is better.

**Review by florimondmanca — COMMENTED** [review]
Thanks for rebasing this against master! Reviewing what this PR has become, I'm not sure this change brings much value in terms of API usability. Eg `httpx.Response(200)` reads quite well as "HTTP 200", while `httpx.Response(status_code=200)` feels a bit _forced_? What do you think?

**Comment by tomchristie** [comment]
I'm going to suggest that we close this one off.

On reassessing, I *don't* think we want it to be keyword-only, and #1227 doesn't actually imply that we should do this. There's nothing wrong with this...

```python
response = httpx.Response(200, json={"hello": "world"})
```

It's *possible* that at some point we might decide to include a *default* value, so that we also support the following...

```python
response = httpx.Response(json={"hello": "world"})
```

But it's not obvious if we do or don't want that, and there's no reason for us to require it for #1227, so let's just pass on it for now.

**Comment by j178** [comment]
Fair enough, I am totally OK with or without this change. Going to close this.

**Pull request closed without merging**